libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -653,12 +653,17 @@ instance HasSpec fn a => Semigroup (Specification fn a) where
       $ MemberSpec
       $ nub
       $ intersect as as'
-  MemberSpec as <> TypeSpec s cant =
-    MemberSpec $
-      nub $
-        filter
-          (flip (conformsTo @fn) s)
-          (filter (`notElem` cant) as)
+  ms@(MemberSpec as) <> ts@TypeSpec {} =
+    case nub $ filter (`conformsToSpec` ts) as of
+      [] ->
+        ErrorSpec
+          ( NE.fromList
+              [ "The two " ++ show (typeRep (Proxy @a)) ++ " Specifications are inconsistent."
+              , "  " ++ show ms
+              , "  " ++ show ts
+              ]
+          )
+      as' -> MemberSpec as'
```

Comment:
This seems to be missing one of the filters  `(filter (`notElem` cant) as) ` - but perhaps on purpose?

Reply:
Oh I see! Makes sense, thank you for explaining.